### ISSUE TITLE
Trouble following packaging libraries tutorial - AttributeError: module 'virtualenv' has no attribute 'cli_run'

### ISSUE THREAD

**Issue by adityaponnada** [pypa/packaging-problems#450]
Thank you for providing feedback on Python packaging!

To help us help you, please fill out as much of the following as you can. If a question is not relevant, feel free to skip it.

1. What is your operating system and version?

Windows 10

2. What is your Python version?

3.7.3

3. What version of pip do you have?

21.0.1

4. If following an online tutorial or guide, please provide a link to the page or section giving you the trouble:

https://packaging.python.org/tutorials/packaging-projects/#generating-distribution-archives

5. Could you describe your issue in as much detail as possible?

I am having trouble with `python -m build ` with the following error.
```
Traceback (most recent call last):
  File "C:\Users\Aditya\Anaconda3\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\Aditya\Anaconda3\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Aditya\Anaconda3\lib\site-packages\build\__main__.py", line 214, in <module>
    main(sys.argv[1:], 'python -m build')
  File "C:\Users\Aditya\Anaconda3\lib\site-packages\build\__main__.py", line 206, in main
    build_package(args.srcdir, outdir, distributions, config_settings, not args.no_isolation, args.skip_dependencies)
  File "C:\Users\Aditya\Anaconda3\lib\site-packages\build\__main__.py", line 94, in build_package
    _build_in_isolated_env(builder, outdir, distributions, config_settings)
  File "C:\Users\Aditya\Anaconda3\lib\site-packages\build\__main__.py", line 52, in _build_in_isolated_env
    with IsolatedEnvBuilder() as env:
  File "C:\Users\Aditya\Anaconda3\lib\site-packages\build\env.py", line 65, in __enter__
    executable, scripts_dir = _create_isolated_env_virtualenv(self._path)
  File "C:\Users\Aditya\Anaconda3\lib\site-packages\build\env.py", line 156, in _create_isolated_env_virtualenv
    result = virtualenv.cli_run(cmd, setup_logging=False)
AttributeError: module 'virtualenv' has no attribute 'cli_run'
```

**Comment by henryiii** [comment]
What version of virtualenv do you have? @gaborbernat is this something known?

**Comment by layday** [comment]
build requires virtualenv 20.0.35 or later.  I presume what happened here is that build was _not_ installed with the `virtualenv` extra but that virtualenv was installed separately.  build does not check that the version of virtualenv is supported but prefers it over venv if it can be imported.

**Comment by henryiii** [comment]
Would it help to do a check with a nicer error message if too old?

**Comment by adityaponnada** [comment]
I just upgraded the 'virtualenv' and seems to be working fine. Just curious, once I have the API token, where do I create the test user name for testpypi?

**Comment by henryiii** [comment]
Register at https://test.pypi.org

**Comment by henryiii** [comment]
Fixed in build 0.3.0 (without the virtualenv manual upgrade)! Please reopen if there are still issues.
